metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -239,7 +244,7 @@ public void deactivate() {
 
         List<ApiMessageAndVersion> records = new ArrayList<>();
         records.add(new ApiMessageAndVersion(record,
-            REGISTER_BROKER_RECORD.highestSupportedVersion()));
+            metadataVersionProvider.activeVersion().recordVersion(REGISTER_BROKER_RECORD)));
```

Comment:
Rather than always picking the highest version, we now consult `MetadataVersion#recordVersion`. We'll need this anytime we construct a record.